lib/inferno/dsl/fhir_evaluation/rules/all_search_parameters_have_examples.rb

```diff
@@ -36,6 +38,19 @@ def check(context)
             context.add_result result
           end
 
+          def fhirpath_is_available?(context)
+            Faraday.new(
+              URI(ENV.fetch('FHIRPATH_URL', nil)), request: { timeout: 600 }
```

Comment:
I'm not sure about this approach -- this is getting into the internals of the fhirpath service, which should stay encapsulated in the FhirpathEvaluation module. Can we just catch the existing exception that gets thrown when the server isn't available and report it in a friendlier way?

Reply:
I thought what we wanted to do was try something simple (ping) then stop there before the rule begins if the server isn't available. My problem was there wasn't a way to simply ping the service provided by the service. Yes I agree with you that this should stay in the module.
What I can do alternatively is to run the rule and if there's a response of exception by unavailable server, halt the rule execution. If that's the case, likely it will be the first fhirpath trial to raise the red flag and stop there, by that we can save unnecessary trials for the rest. I will try and come back.

Reply:
So I made changes to simply handle the exceptions; error if it's no connection to server, warning if it's something about fhirpath evaluation.

https://github.com/inferno-framework/inferno-core/pull/667/commits/37d0342b7d10374b5a175b423924212b8f7e5f27